cmd/podman/containers/exec.go

```diff
@@ -84,6 +86,10 @@ func execFlags(cmd *cobra.Command) {
 	flags.StringVarP(&execOpts.WorkDir, workdirFlagName, "w", "", "Working directory inside the container")
 	_ = cmd.RegisterFlagCompletionFunc(workdirFlagName, completion.AutocompleteDefault)
 
+	waitFlagName := "wait"
```

Comment:
We also need to add docs.

Reply:
ack on tests. But if it's hidden, does it need docs?

Reply:
@vrothberg @Luap99 @rhatdan would the --wait flag addition be better off in a separate PR?

Reply:
SGTM, thanks!

Reply:
Do we want to support --wait outside of podmansh? I would say no for now so no docs and hidden IMO.

Reply:
No support yet.  There could be use cases, but I believe we should hide it until we have a true use case. Might be something for testing though @edsantiago WDYT?

Reply:
Sorry... I've been completely unable to review this PR: huge + moving quickly. I don't understand `wait` enough to comment.

Reply:
added best effort system and integration tests :) for --wait . PTAL.

Reply:
The basic idea for tests would be something like

```
podman run --name test1 -d image 
podman exec --wait 10 test1 echo hello
```
Podman exec will attempt to exec into the container as soon as it can, but will exponentially fall back for up to 10 seconds before failing, if the container is not running.